src/Framework/NativeMethods.cs

```diff
@@ -1296,26 +1298,27 @@ internal static int GetParentProcessId(int processId)
         else
 #endif
         {
-            SafeProcessHandle hProcess = OpenProcess(eDesiredAccess.PROCESS_QUERY_INFORMATION, false, processId);
-
-            if (!hProcess.IsInvalid)
+            using SafeProcessHandle hProcess = OpenProcess(eDesiredAccess.PROCESS_QUERY_INFORMATION, false, processId);
```

Comment:
There is a finally block with an explicit `Dispose` call down below. I think the try/catch can now be deleted.